clients/database/request_metric.go

```diff
@@ -27,6 +27,7 @@ type ProxiedRequestMetric struct {
 	UserAgent                   *string
 	Referer                     *string
 	Origin                      *string
+	ResponseBackend             string
```

Comment:
prefer pointer to string to allow for differentiating between "value is empty because it was unset / nil, vs the empty value is what was set"

Reply:
the database has a default value of "DEFAULT" for the column.

realizing this doesn't work for requests that fail (ie. if they get 502'd because they aren't in the map). will update column to allow null and make this a pointer.

i think i'll also abandon backfilling the column to "DEFAULT" unless you think that's still worthwhile?

Reply:
digging deeper, it looks like we aren't creating metrics for failing requests.

the only case in which the `response_backend` wouldn't be correct if it defaulted to "DEAFULT" is in the case where the host is not in the host -> backend url map. in that case we short-circuit respond 502:
https://github.com/Kava-Labs/kava-proxy-service/blob/ad18471499a79cd693f683d0216082f1d50a7fde/service/middleware.go#L172-L180

when we `return` there, we forego adding details to the context. this means for all the context values, casting to string will fail (`ok == false`). then we return early, skipping metric creation.

some examples:
https://github.com/Kava-Labs/kava-proxy-service/blob/ad18471499a79cd693f683d0216082f1d50a7fde/service/middleware.go#L331-L356

all those values are set after the failing request would short-circuit the middleware with the 502.

so the take away:
1. we aren't collecting failing request metrics. we probably should? though i think they'd just be indicative of a configuration problem (either an unintended host is pointing to the proxy that we didn't setup, or we didn't setup an intended host)
2. the request metrics table is currently only for requests that were properly routed. the repsponse_backend should always be set and so a string is more appropriate than a pointer. additionally, the backfill of existing metrics (and default column value) is valid.